File: Backend/src/main/java/com/example/demo/entity/Product.java
package com.example.demo.entity;

import lombok.*;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.LinkedList;
import java.util.List;
import java.util.Set;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Long id;
    private String name;
//    @OneToMany(mappedBy = "product", cascade = CascadeType.PERSIST)
//    private final List<OrderProduct> product = new ArrayList<>();
    private double cost;
    private String description;
    private int quantity;
    @Lob
    private String image;
    private String category;
}

File: Frontend/src/app/entity/Product.ts
export class Product {
  id: number
  name: string
  product?: []
  cost: string
  description?: string
  quantity?: number
  image: string
  category?: string

  constructor(id: number, name: string, product: [], cost: string, description: string, quantity: number, image: string, category: string) {
    this.id = id;
    this.name = name;
    this.product = product;
    this.cost = cost;
    this.description = description;
    this.quantity = quantity;
    this.image = image;
    this.category = category;
  }

}
